src/app/ui/editor/select_box_state.cpp

```diff
@@ -250,47 +297,238 @@ void SelectBoxState::postRenderDecorator(EditorPostRender* render)
+        }
+        else {
+          // Just VGrid Active, HGrid disabled:
+          // Bottom band
+          if (sp.y2() < vp.y2())
+            render->fillRect(dark, gfx::Rect(rc.x, sp.y2(), vp.x2()-rc.x, vp.y2()-sp.y2()));
+          // Right band
+          if (rc.x2() < vp.x2())
+            render->fillRect(dark, gfx::Rect(rc.x2(), rc.y, vp.x2()-rc.x2(), sp.y2()-rc.y));
+        }
+      }
+      else {
+        if (hasFlag(Flags::HGrid)) {
+          // Just HGrid Active, VGrid disabled
+          // Bottom band
+          if (rc.y2() < vp.y2())
+            render->fillRect(dark, gfx::Rect(rc.x, rc.y2(), vp.x2()-rc.x, vp.y2()-rc.y2()));
+          // Right band
+          if (sp.x2() < vp.x2())
+            render->fillRect(dark, gfx::Rect(sp.x2(), rc.y, vp.x2()-sp.x2(), rc.h));
+        }
+      }
 
-  if (hasFlag(Flags::Grid)) {
-    if (rc.w > 0) {
-      for (int x=rc.x+rc.w*2; x<=sp.x+sp.w; x+=rc.w)
-        render->drawLine(gridColor, x, rc.y, x, sp.y+sp.h);
+      // Draw dark zones
+      if (hasFlag(Flags::PaddingRulers)) {
+        // Draw dark zones when Padding is present
+        if (hasFlag(Flags::VGrid) && hasFlag(Flags::HGrid)) {
+          // Draw vertical dark padding big bands
+          std::vector<int> padXTips;
+          std::vector<int> padYTips;
+          if (rc.w > 0) {
+            for (int x=rc.x+rc.w; x<=sp.x+sp.w; x+=(rc.w+padding.w)) {
+              if (x + padding.w > sp.x2())
+                render->fillRect(dark, gfx::Rect(x, rc.y, sp.x2()-x, sp.h-rc.y));
+              else
+                render->fillRect(dark, gfx::Rect(x, rc.y, padding.w, sp.h-rc.y));
+              padXTips.push_back(x);
+            }
+          }
+          if (rc.h > 0) {
+            // Draw horizontal dark chopped padding bands (to complete the dark grid)
+            for (int y=rc.y+rc.h; y<=sp.y+sp.h; y+=(rc.h+padding.h)) {
+              for (const int& padXTip : padXTips) {
+                if (y+padding.h > sp.y2())
+                  render->fillRect(dark, gfx::Rect(padXTip-rc.w, y, rc.w, sp.y2()-y));
+                else
+                  render->fillRect(dark, gfx::Rect(padXTip-rc.w, y, rc.w, padding.h));
+              }
+              if (!padXTips.empty()) {
+                if (padXTips.back() + padding.w < sp.x2()) {
+                  if (y + padding.h > sp.y2())
+                    render->fillRect(dark, gfx::Rect( padXTips.back() + padding.w
+                                                    , y
+                                                    , sp.x2() - padXTips.back() - padding.w
+                                                    , sp.y2() - y));
+                  else
+                    render->fillRect(dark, gfx::Rect( padXTips.back() + padding.w
+                                                    , y
+                                                    , sp.x2() - padXTips.back() - padding.w
+                                                    , padding.h));
```

Comment:
Minor change: I don't like the style of these lines 😬they could be just:
```
                if (padXTips.back() + padding.w < sp.x2()) {
                  if (y + padding.h > sp.y2())
                    render->fillRect(dark, gfx::Rect(padXTips.back() + padding.w, y,
                                                     sp.x2() - padXTips.back() - padding.w,
                                                     sp.y2() - y));
                  else
                    render->fillRect(dark, gfx::Rect(padXTips.back() + padding.w, y,
                                                     sp.x2() - padXTips.back() - padding.w,
                                                     padding.h));
                }
``` 
Same for other cases below.